src/main/java/org/utplsql/cli/ReporterManager.java

```diff
@@ -15,42 +16,45 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
 
 class ReporterManager {
 
     private List<ReporterOptions> reporterOptionsList;
     private List<Throwable> reporterGatherErrors;
     private ExecutorService executorService;
 
-    ReporterManager(List<String> reporterParams ) {
-        initReporterOptionsList(reporterParams);
+    ReporterManager(ReporterConfig[] reporterConfigs ) {
+        reporterOptionsList = new ArrayList<>();
+        if ( reporterConfigs != null && reporterConfigs.length > 0 ) {
+            loadOptionsFromConfigs( reporterConfigs );
+        }
+        else {
+            reporterOptionsList.add(getDefaultReporterOption());
+        }
     }
 
-    private void initReporterOptionsList( List<String> reporterParams ) {
-        reporterOptionsList = new ArrayList<>();
-        ReporterOptions reporterOptions = null;
+    private void loadOptionsFromConfigs( ReporterConfig[] reporterConfigs ) {
+        boolean printToScreen = false;
+        for (ReporterConfig reporterConfig : reporterConfigs) {
+            ReporterOptions option = new ReporterOptions(
+                    reporterConfig.getName(),
+                    reporterConfig.getOutput());
 
-        for (String p : reporterParams) {
-            if (reporterOptions == null || !p.startsWith("-")) {
-                reporterOptions = new ReporterOptions(p);
-                reporterOptionsList.add(reporterOptions);
-            }
-            else
-            if (p.startsWith("-o=")) {
-                reporterOptions.setOutputFileName(p.substring(3));
-            }
-            else
-            if (p.equals("-s")) {
-                reporterOptions.forceOutputToScreen(true);
-            }
-        }
+            option.forceOutputToScreen(reporterConfig.isForceToScreen());
+            reporterOptionsList.add(option);
 
-        // If no reporter parameters were passed, use default reporter.
-        if (reporterOptionsList.isEmpty()) {
-            reporterOptionsList.add(new ReporterOptions(CoreReporters.UT_DOCUMENTATION_REPORTER.name()));
+            // Check printToScreen validity
+            if (option.outputToScreen() && printToScreen)
+                throw new IllegalArgumentException("You cannot configure more than one reporter to output to screen");
+            printToScreen = option.outputToScreen();
         }
     }
 
+    private ReporterOptions getDefaultReporterOption() {
+        return new ReporterOptions(CoreReporters.UT_DOCUMENTATION_REPORTER.name());
+    }
+
     private void abortGathering(Throwable e) {
         addGatherError(e);
         executorService.shutdownNow();
```

Comment:
I realized this might be an issue. We get the executorService from the outside but we shut it down here. I'm concerned if the caller is ok with its executorService been shut down.

Reply:
That's actually the purpose: If one Reporter Consumption fails, it should abort all the other Reporter threads.
But that's one of the things I want to do differently/more clear

Reply:
lets abort tasks then, not the excutor

Reply:
Yes, will make that more clear in a restructured version